app/controllers/vacancies_controller.rb

```diff
@@ -4,7 +4,7 @@ class VacanciesController < ApplicationController
 
   def index
     @vacancies_search = Search::VacancySearch.new(form.to_hash, sort: form.sort)
-    @pagy, @vacancies = pagy(@vacancies_search.vacancies)
+    @pagy, @vacancies = pagy(@vacancies_search.vacancies, count: @vacancies_search.total_count)
```

Comment:
It prevents `Pagy` from triggering an expensive extra full query with COUNT internally for the pagination menu. When `count` is given, use the given value.

In this case, we pass a `total_count` value that has been memoised so that further calls won't trigger the query again.